moshi-proguard-rule-gen/src/main/kotlin/dev/zacsweers/moshix/proguardgen/MoshiProguardGenSymbolProcessor.kt

```diff
@@ -131,7 +131,8 @@ public class MoshiProguardGenSymbolProcessor(private val environment: SymbolProc
                 val adapterCanonicalName =
                   ClassName(targetType.packageName, adapterName).canonicalName
                 for (target in nestedSealedClassNames.sorted()) {
-                  writer.appendLine("-if class $target")
+                  val targetReflectionName = target.reflectionName()
+                  writer.appendLine("-if class $targetReflectionName")
                   writer.appendLine("-keep class $adapterCanonicalName {")
```

Comment:
We probably actually want to use a reflection name here too right?

Reply:
Looks like that would sense, yes. Though I guess generated adapters are always top-level classes? And looking at the generated rules, `canonicalName` does not seem to include backticks around keywords like `toString()` did:

```
-if class com.example.app.`data`.model.Message.Success
-keep class com.example.app.data.model.MessageJsonAdapter {
    public <init>(com.squareup.moshi.Moshi);
}
```
But there might be other cases where I'm not thinking about right now... Will update!

Reply:
Yeah doesn't hurt to be safe. They are indeed always top-level though